tests/storage/datavolume.go

```diff
@@ -116,8 +116,29 @@ var _ = SIGDescribe("DataVolume Integration", func() {
 			if !volumeExpansionAllowed {
 				Skip("Skip when volume expansion storage class not available")
 			}
-			vmi, dataVolume := tests.NewRandomVirtualMachineInstanceWithDisk(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros), testsuite.GetTestNamespace(nil), sc, k8sv1.ReadWriteOnce, volumeMode)
-			tests.AddUserData(vmi, "cloud-init", "#!/bin/bash\necho 'hello'\n")
+
+			imageUrl := cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros)
+			var err error
+
+			dataVolume := libdv.NewDataVolume(
+				libdv.WithRegistryURLSourceAndPullMethod(imageUrl, cdiv1.RegistryPullNode),
+				libdv.WithPVC(
+					libdv.PVCWithStorageClass(sc),
+					libdv.PVCWithVolumeSize(tests.DvSizeBySourceURL(imageUrl)),
+					libdv.PVCWithAccessMode(k8sv1.ReadWriteOnce),
+					libdv.PVCWithVolumeMode(volumeMode),
+				),
+			)
+			dataVolume, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(testsuite.GetTestNamespace(nil)).Create(context.Background(), dataVolume, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			libstorage.EventuallyDV(dataVolume, 240, Or(HaveSucceeded(), BeInPhase(cdiv1.WaitForFirstConsumer), BeInPhase(cdiv1.PendingPopulation)))
+
+			vmi := libvmi.New(
+				libvmi.WithDataVolume("disk0", dataVolume.Name),
+				libvmi.WithResourceMemory("1Gi"),
+				libvmi.WithCloudInitNoCloudUserData("#!/bin/bash\necho 'hello'\n"),
```

Comment:
We should do something about this bash hello.

Reply:
See discussion in https://github.com/kubevirt/kubevirt/pull/11370

Reply:
@RoniKishner  can you at least copy the code from https://github.com/kubevirt/kubevirt/blob/ee87726a579c44609316a5d35d7dca1b7da94c87/tests/libvmi/factory.go#L51 ?
It would make it clearer for a reader of your code to better understand why this is needed.

Reply:
Please note my request and consider it.

Reply:
I changed to line to use `WithCloudInitNoCloudEncodedUserData("#!/bin/bash\necho hello\n")`

Reply:
imho using a named `withNonEmptyUserData` would make it much clearer *why* you're using this awkward string.